### PR TITLE
Feature/core 548 twin onboarding

### PR DESCRIPTION
**Pull request by Tschuck**

### CODE REVIEW

File: dapps/evancore.vue.libs/src/components/forms/form/form.ts [hunk 1]
```diff
@@ -211,20 +211,40 @@ export default class EvanFormComponent extends mixins(EvanComponent) {
...
+        return this.$t(control[type]);
+      } else if (!control[type]) {
+        return control[type];
       }
     }
 
     // return directly specified translation
-    if (control.uiSpecs && control.uiSpecs.attr && control.uiSpecs.attr[type]) {
-      return this.$t(control.uiSpecs.attr[type]);
+    let returnTranslation = type !== 'hint';
```
Comment (karladler) [inline dapps/evancore.vue.libs/src/components/forms/form/form.ts:222]:
```suggestion
    let returnTranslation = type !== 'hint';
    
```

File: dapps/evancore.vue.libs/src/components/forms/form/form.ts [hunk 2]
```diff
@@ -211,20 +211,40 @@ export default class EvanFormComponent extends mixins(EvanComponent) {
   getTranslation(control: EvanFormControl, type: string) {
     // if manual error text was specified, translate it and return it directly
     if (type === 'error') {
-      if (typeof control.error !== 'boolean') {
-        return this.$t(control.error);
-      } else if (!control.error) {
-        return control.error;
+      if (typeof control[type] !== 'boolean') {
```
Comment (karladler) [inline dapps/evancore.vue.libs/src/components/forms/form/form.ts]: why change it and lose TS support, when type is always checked before if it's === 'error' ?

File: dapps/evancore.vue.libs/src/components/forms/form/form.ts [hunk 3]
```diff
@@ -211,20 +211,40 @@ export default class EvanFormComponent extends mixins(EvanComponent) {
...
-    if (control.uiSpecs && control.uiSpecs.attr && control.uiSpecs.attr[type]) {
-      return this.$t(control.uiSpecs.attr[type]);
+    let returnTranslation = type !== 'hint';
+    if (control.uiSpecs && control.uiSpecs.attr &&
+        control.uiSpecs.attr.hasOwnProperty(type)) {
+      // if it's a hint, check if it's set to true or string
+      if (type === 'hint') {
+        if (control.uiSpecs.attr.hint) {
+          // if true, enable hint translation
+          if (typeof control.uiSpecs.attr.hint === 'boolean') {
```
Comment (karladler) [inline dapps/evancore.vue.libs/src/components/forms/form/form.ts:229]: comments says if true, but if would also return true if it's false ;)

File: dapps/evancore.vue.libs/src/components/forms/form/form.ts [hunk 4]
```diff
@@ -211,20 +211,40 @@ export default class EvanFormComponent extends mixins(EvanComponent) {
...
+        return this.$t(control.uiSpecs.attr[type]);
+      }
     }
 
-    // return default translation
-    return this.$t(`${ this.i18nScope }.${ control.name }.${ type }`);
+    if (returnTranslation) {
+      // return default translation
+      return this.$t(`${ this.i18nScope }.${ control.name }.${ type }`);
+    } else {
```
Comment (karladler) [inline dapps/evancore.vue.libs/src/components/forms/form/form.ts:245]: else { may be omitted when returned in previous if branch (early-return)

File: dapps/evancore.vue.libs/src/components/forms/input/input.ts [hunk 5]
```diff
@@ -31,4 +31,11 @@ import EvanControlComponent from '../control/control';
  * @selector      evan-form-control-input
  */
 @Component({})
-export default class InputComponent extends mixins(EvanControlComponent) { }
+export default class InputComponent extends mixins(EvanControlComponent) {
+  /**
+   * Input type attribute
+   */
+  @Prop({
+    default: 'text',
```
Comment (karladler) [inline dapps/evancore.vue.libs/src/components/forms/input/input.ts:39]: default is 'text' but type is boolean?

File: dapps/evancore.vue.libs/src/components/forms/textarea/textarea.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see http://www.gnu.org/licenses/ or
+  write to the Free Software Foundation, Inc., 51 Franklin Street,
+  Fifth Floor, Boston, MA, 02110-1301 USA, or download the license from
+  the following URL: https://evan.network/license/
+*/
+
+// vue imports
+import Component, { mixins } from 'vue-class-component';
+import Vue from 'vue';
```
Comment (karladler) [inline dapps/evancore.vue.libs/src/components/forms/textarea/textarea.ts]:
```suggestion
```

File: dapps/evancore.vue.libs/src/components/forms/textarea/textarea.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import { Prop, Watch } from 'vue-property-decorator';
```
Comment (karladler) [inline dapps/evancore.vue.libs/src/components/forms/textarea/textarea.ts]:
```suggestion
```

File: dapps/evancore.vue.libs/src/components/forms/textarea/textarea.vue [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      :value="value"
+      @blur="$emit('blur')"
+      @focus="$parent.$emit('setFocus')"
+      @input="$emit('input', $event.target.value)"
+    />
+  </evan-form-control>
+</template>
+
+<script lang="ts">
+  import Component from './textarea'
```
Comment (karladler) [inline dapps/evancore.vue.libs/src/components/forms/textarea/textarea.vue:35]: 👍

**Review by karladler — CHANGES_REQUESTED** [review]
just a few small things 8see comments) and may be merged then.

File: dapps/evancore.vue.libs/src/components/forms/form/form.ts (hunk 3 above)
Comment (Tschuck) [reply dapps/evancore.vue.libs/src/components/forms/form/form.ts:229]: previous if will handle the false ;)

**Pull request merged**